extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,103 @@
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, clearData.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    String encrypt(String keyName, ClearData clearData, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either a ciphertext or an error.
+     * Each element can specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<String> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
+
+    ClearData decrypt(String keyName, String ciphertext);
+
+    ClearData decrypt(String keyName, String ciphertext, TransitContext transitContext);
+
+    List<ClearData> decrypt(String keyName, List<DecryptionRequest> requests);
+
+    // rewrap
+
+    String rewrap(String keyName, String ciphertext);
+
+    String rewrap(String keyName, String ciphertext, TransitContext transitContext);
+
+    List<String> rewrap(String keyName, List<RewrappingRequest> requests);
+
+    // sign
+
+    String sign(String keyName, String input);
+
+    String sign(String keyName, SignatureInput input, TransitContext transitContext);
+
+    List<String> sign(String keyName, SigningBatchRequest batchRequest);
+
+    // verify
+
+    boolean verify(String keyName, String signature, String input);
+
+    boolean verify(String keyName, String signature, SignatureInput input, TransitContext transitContext);
```

Comment:
@vsevel would it make sense to collapse 2nd and 3rd parameters into `VerificationRequest` ?

Reply:
I would leave both params because both of them form the core of the service, and it shows clearly what we are doing: _verify_ _signature_ of _input_, and it is consistent with the other services where we do not use request objects for non batched use cases.
but I do not feel too strong about it. if you really think it is appropriate we can do it.

Reply:
@vsevel it sounds reasonable, we can do a quick revisiting of it later